homeassistant/components/music_assistant/media_player.py

```diff
@@ -0,0 +1,729 @@
+            # resolve HA entity_id to MA player_id
+            if (hass_state := self.hass.states.get(child_entity_id)) is None:
+                continue
+            if (mass_player_id := hass_state.attributes.get("mass_player_id")) is None:
+                continue
+            player_ids.append(mass_player_id)
+        await self.mass.players.cmd_sync_many(self.player_id, player_ids)
+
+    @catch_musicassistant_error
+    async def async_unjoin_player(self) -> None:
+        """Remove this player from any group."""
+        await self.mass.players.player_command_unsync(self.player_id)
+
+    @catch_musicassistant_error
+    async def _async_play_media_advanced(
+        self,
+        media_id: list[str],
+        artist: str | None = None,
+        album: str | None = None,
+        enqueue: MediaPlayerEnqueue | QueueOption | None = None,
+        radio_mode: bool | None = None,
+        media_type: str | None = None,
+    ) -> None:
+        """Send the play_media command to the media player."""
+        # pylint: disable=too-many-arguments
+        media_uris: list[str] = []
+        item: MediaItemType | None = None
+        # work out (all) uri(s) to play
+        for media_id_str in media_id:
+            # URL or URI string
+            if "://" in media_id_str:
+                media_uris.append(media_id_str)
+                continue
+            # try content id as library id
+            if media_type and media_id_str.isnumeric():
+                with suppress(MediaNotFoundError):
+                    if media_type is not None:
+                        continue
+                    item = await self.mass.music.get_item(  # type: ignore[unreachable]
+                        MediaType(media_type), media_id_str, "library"
+                    )
+                    if item.uri is not None:
+                        continue
+                    media_uris.append(item.uri)
+                    continue
+            # try local accessible filename
+            elif await asyncio.to_thread(os.path.isfile, media_id_str):
+                media_uris.append(media_id_str)
+                continue
+            # last resort: lookup by name/search
+            if item := await self._get_item_by_name(
+                media_id_str, artist, album, media_type
+            ):
+                if TYPE_CHECKING:
+                    assert item.uri is not None
+                media_uris.append(item.uri)
+
+        if not media_uris:
+            raise MediaNotFoundError(
+                f"Could not resolve {media_id} to playable media item"
+            )
+
+        # determine active queue to send the play request to
+        if TYPE_CHECKING:
+            assert self.player.active_source is not None
+        if queue := self.mass.player_queues.get(self.player.active_source):
+            queue_id = queue.queue_id
+        else:
+            queue_id = self.player_id
+        if TYPE_CHECKING:
+            assert radio_mode is not None
```

Comment:
if we want it not to be None, why do we type it as such?

Reply:
This check is incorrect - radio mode is an optional argument and thus can be None

Reply:
Unfortunately in the client code it is not optional, but has a default of false
I'll check for None, and set to False if None to match the client, but we should change it in the client to be optional I suppose